src/ui/SubredditPage.tsx

```diff
@@ -19,27 +19,42 @@ export default class SubredditPage extends Page implements common.SubredditView
   private _items: RedditPost[] = [];
   private _mode: ViewMode;
   private loading: boolean;
-  @getById private collectionView: CollectionView;
+  @getById private listView: ListView<RedditPost>;
 
   constructor(properties?: Properties<SubredditPage>) {
     super(properties);
     this.append(
-      <CollectionView id='collectionView'
+      <ListView
+          id='listView'
           stretch
           background='#f5f5f5'
-          cellType={() => this._mode}
-          cellHeight={(_index, type) => isList(type) ? 96 : 160}
-          createCell={(type: string) => this.createCell(type)}
-          updateCell={(view, index) => (view as Cell).item = this._items[index].data}
-          onLastVisibleIndexChanged={this.handleLastVisibleIndexChanged}/>
+          onLastVisibleIndexChanged={({value})=> this.handleLastVisibleIndexChanged(value)}
```

Comment:
You should be able to directly attach the listener as before. There reason i prefere that i find JS code within an attribute hard to read.